controllers/toolchaincluster/healthchecker.go

```diff
@@ -20,94 +15,19 @@ const (
 	clusterReachableMsg    = "cluster is reachable"
 )
 
-type HealthChecker struct {
-	localClusterClient     client.Client
-	remoteClusterClient    client.Client
-	remoteClusterClientset *kubeclientset.Clientset
-	logger                 logr.Logger
-}
-
-func (hc *HealthChecker) updateIndividualClusterStatus(ctx context.Context, toolchainCluster *toolchainv1alpha1.ToolchainCluster) error {
-
-	currentClusterStatus := hc.getClusterHealthStatus(ctx)
-
-	for index, currentCond := range currentClusterStatus.Conditions {
-		for _, previousCond := range toolchainCluster.Status.Conditions {
-			if currentCond.Type == previousCond.Type && currentCond.Status == previousCond.Status {
-				currentClusterStatus.Conditions[index].LastTransitionTime = previousCond.LastTransitionTime
-			}
-		}
-	}
+// getClusterHealth gets the kubernetes cluster health status by requesting "/healthz"
+func getClusterHealthStatus(ctx context.Context, remoteClusterClientset *kubeclientset.Clientset) (bool, error) {
 
-	toolchainCluster.Status = *currentClusterStatus
-	if err := hc.localClusterClient.Status().Update(ctx, toolchainCluster); err != nil {
-		return errors.Wrapf(err, "Failed to update the status of cluster %s", toolchainCluster.Name)
-	}
-	return nil
-}
-
-// getClusterHealthStatus gets the kubernetes cluster health status by requesting "/healthz"
-func (hc *HealthChecker) getClusterHealthStatus(ctx context.Context) *toolchainv1alpha1.ToolchainClusterStatus {
-	clusterStatus := toolchainv1alpha1.ToolchainClusterStatus{}
-	body, err := hc.remoteClusterClientset.DiscoveryClient.RESTClient().Get().AbsPath("/healthz").Do(ctx).Raw()
+	lgr := log.FromContext(ctx)
+	body, err := remoteClusterClientset.DiscoveryClient.RESTClient().Get().AbsPath("/healthz").Do(ctx).Raw()
 	if err != nil {
-		hc.logger.Error(err, "Failed to do cluster health check for a ToolchainCluster")
-		clusterStatus.Conditions = append(clusterStatus.Conditions, clusterOfflineCondition())
+		lgr.Error(err, "Failed to do cluster health check for a ToolchainCluster")
+		return false, err
 	} else {
 		if !strings.EqualFold(string(body), "ok") {
-			clusterStatus.Conditions = append(clusterStatus.Conditions, clusterNotReadyCondition(), clusterNotOfflineCondition())
+			return false, nil
 		} else {
-			clusterStatus.Conditions = append(clusterStatus.Conditions, clusterReadyCondition())
+			return true, nil
 		}
 	}
```

Comment:
if you like, this can be simplified as follows:

```go
	if err != nil {
		lgr.Error(err, "Failed to do cluster health check for a ToolchainCluster")
		return false, err
	}

	return strings.EqualFold(string(body), "ok"), nil
```